### PR TITLE
Update abieos

### PR DESCRIPTION
**Pull request by jgiszczak**
Upgrade abieos, providing nodeos KV support.

### CODE REVIEW

File: src/state_history_pg.hpp [hunk 1]
```diff
@@ -145,48 +150,75 @@ inline std::string sql_str(bool bulk, uint32_t v)
...
-inline std::string sql_str(bool bulk, transaction_status v)                             { return quote(bulk, to_string(v)); }
-inline std::string sql_str(bool bulk, abieos::symbol v)                                 { return quote(bulk, abieos::symbol_to_string(v.value)); }
+inline std::string sql_str(bool bulk, eosio::varuint32 v)                               { return std::to_string(v.value); }
+inline std::string sql_str(bool bulk, eosio::varint32 v)                                { return std::to_string(v.value); }
+#ifndef ABIEOS_NO_INT128
+inline std::string sql_str(bool bulk, const abieos::int128& v)                          { auto nv = -v; return v < 0 ? std::string("-") + abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(nv)).get_array()) : abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
+inline std::string sql_str(bool bulk, const abieos::uint128& v)                         { return abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
```
Comment (tbfleming) [inline src/state_history_pg.hpp:157]: abieos's `uint128` and `int128` structs define a cast to `std::string`

File: src/state_history_pg.hpp [hunk 2]
```diff
@@ -251,83 +289,96 @@ inline std::string empty_to_sql<abieos::bytes>(pqxx::connection&, bool bulk) {
...
     return quote_bytea(bulk, "");
 }
 
 inline abieos::time_point sql_to_time_point(std::string s) {
     if (s.empty())
         return {};
     std::replace(s.begin(), s.end(), ' ', 'T');
-    return abieos::string_to_time_point(s);
+    abieos::time_point tp;
+    auto ts = eosio::json_token_stream(s.data());
```
Comment (tbfleming) [inline src/state_history_pg.hpp:315]: `s` isn't JSON

File: src/state_history_pg.hpp [hunk 3]
```diff
@@ -251,83 +289,96 @@ inline std::string empty_to_sql<abieos::bytes>(pqxx::connection&, bool bulk) {
...
+    from_json(tp, ts);
+    return tp;
+}
+
+template <typename T>
+inline eosio::time_point convert_from_string(std::string_view s) {
```
Comment (tbfleming) [inline src/state_history_pg.hpp:321]: Function isn't dependent on `T`; it blindly assumes `time_point`

File: src/state_history_pg.hpp (hunk 1 above)
Comment (jgiszczak) [reply src/state_history_pg.hpp:157]: In Clang on Ubuntu, `__int128` is available but corresponding standard library functions to convert it don't seem to be.  Current CMake behavior does not define `ABIEOS_NO_INT128`, so `struct uint128` and `struct int128` are not available.  Should that be changed?  Neither of them in abieos.hpp provides a `std::string` operator.

Comment (tbfleming) [reply src/state_history_pg.hpp:157]: Got it. In that case, you might want to create separate conversion functions for those. Also, that `reinterpret_cast` is a strict-aliasing violation for both the `__int128` and `struct int128` cases.

File: src/state_history_pg.hpp (hunk 3 above)
Comment (jgiszczak) [reply src/state_history_pg.hpp]: Corrected.

File: src/state_history_pg.hpp (hunk 2 above)
Comment (jgiszczak) [reply src/state_history_pg.hpp]: Properly use template specialization conversion function previously added for block timestamp.

File: src/state_history_pg.hpp [hunk 4]
```diff
@@ -166,8 +166,8 @@ inline std::string sql_str(bool bulk, int64_t v)
...
 inline std::string sql_str(bool bulk, eosio::varint32 v)                                { return std::to_string(v.value); }
 #ifndef ABIEOS_NO_INT128
-inline std::string sql_str(bool bulk, const abieos::int128& v)                          { auto nv = -v; return v < 0 ? std::string("-") + abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(nv)).get_array()) : abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
-inline std::string sql_str(bool bulk, const abieos::uint128& v)                         { return abieos::binary_to_decimal(eosio::fixed_bytes<16, uint8_t>(reinterpret_cast<const uint8_t(&)[16]>(v)).get_array()); }
+inline std::string sql_str(bool bulk, const abieos::int128& v)                          { std::array<uint8_t, 128/8> t; auto nv = -v; return v < 0 ? std::string("-") + abieos::binary_to_decimal(eosio::fixed_bytes<128/8, uint8_t>([&](){std::memcpy(&t, &nv, sizeof(uint8_t)*128/8); return t;}()).get_array()) : abieos::binary_to_decimal(eosio::fixed_bytes<128/8, uint8_t>([&](){std::memcpy(&t, &v, sizeof(uint8_t)*128/8); return t;}()).get_array()); }
```
Comment (tbfleming) [inline src/state_history_pg.hpp:169]: Why go from array to fixed_bytes to array?

Why these in a lambda?

File: src/state_history_pg.hpp (hunk 1 above)
Comment (jgiszczak) [reply src/state_history_pg.hpp]: Added converters for non-native integers.

**Review by tbfleming — APPROVED** [review]

**Pull request merged**